From f44bf34c67b5cbd91eaee8b1b0698df9e1596379 Mon Sep 17 00:00:00 2001
From: Harini Katakam <harini.katakam@xilinx.com>
Date: Mon, 7 Apr 2014 19:46:55 +0530
Subject: [PATCH 346/793] spi: zynq-qspi: num-chipselect property cleanup

Rename num-chipselect property to the commonly used num-cs.
When reading num-cs read into temporary u32 variable and then
copy to master->num_chipselect which is u16.
When num-cs is not present, use a default value instead of
returning error.

Signed-off-by: Harini Katakam <harinik@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/spi/spi-zynq-qspi.c |   16 ++++++++++------
 1 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/spi/spi-zynq-qspi.c b/drivers/spi/spi-zynq-qspi.c
index 8d55ba7..86f4aad 100644
--- a/drivers/spi/spi-zynq-qspi.c
+++ b/drivers/spi/spi-zynq-qspi.c
@@ -121,6 +121,9 @@
  */
 #define MODEBITS			(SPI_CPOL | SPI_CPHA)
 
+/* Default number of chip selects */
+#define ZYNQ_QSPI_DEFAULT_NUM_CS	1
+
 /**
  * struct zynq_qspi - Defines qspi driver instance
  * @regs:		Virtual address of the QSPI controller registers
@@ -628,6 +631,7 @@ static int zynq_qspi_probe(struct platform_device *pdev)
 	struct spi_master *master;
 	struct zynq_qspi *xqspi;
 	struct resource *res;
+	u32 num_cs;
 
 	master = spi_alloc_master(&pdev->dev, sizeof(*xqspi));
 	if (master == NULL)
@@ -691,12 +695,12 @@ static int zynq_qspi_probe(struct platform_device *pdev)
 	/* QSPI controller initializations */
 	zynq_qspi_init_hw(xqspi);
 
-	ret = of_property_read_u32(pdev->dev.of_node, "num-chip-select",
-				   (u32 *)&master->num_chipselect);
-	if (ret < 0) {
-		dev_err(&pdev->dev, "couldn't determine num-chip-select\n");
-		goto clk_dis_all;
-	}
+	ret = of_property_read_u32(pdev->dev.of_node, "num-cs",
+				   &num_cs);
+	if (ret < 0)
+		master->num_chipselect = ZYNQ_QSPI_DEFAULT_NUM_CS;
+	else
+		master->num_chipselect = num_cs;
 
 	master->setup = zynq_qspi_setup;
 	master->set_cs = zynq_qspi_chipselect;
-- 
1.7.5.4

